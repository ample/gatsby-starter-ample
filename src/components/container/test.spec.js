import React from "react"
import renderer from "react-test-renderer"

import { component as Container, fixtures } from "."

describe("Container", () => {
  it("does not cause errors without props", () => {
    const tree = renderer.create(<Container {...fixtures.empty} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders sub-components", () => {
    const tree = renderer.create(<Container {...fixtures.default} />).toJSON()
    expect(tree).toMatchSnapshot()
    const button = tree.children[0].children[0]
    expect(button.type).toBe("a")
    expect(button.children[0]).toBe("Hello World")
  })
})
