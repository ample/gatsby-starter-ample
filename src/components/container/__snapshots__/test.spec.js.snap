// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container does not cause errors without props 1`] = `
<div
  className="container"
>
  <div
    className="content"
  />
</div>
`;

exports[`Container renders sub-components 1`] = `
<div
  className="container"
>
  <div
    className="content"
  >
    <a
      className="button theme_default"
      href="/"
    >
      Hello World
    </a>
  </div>
</div>
`;
